sql/core/src/test/java/test/org/apache/spark/sql/JavaDatasetSuite.java

```diff
@@ -1504,6 +1574,54 @@ public void testSerializeNull() {
     Assert.assertEquals(beans, ds2.collectAsList());
   }
 
+  @Test
+  public void testNonNullField() {
+    NestedSmallBeanWithNonNullField bean1 = new NestedSmallBeanWithNonNullField();
+    SmallBean smallBean = new SmallBean();
+    bean1.setNonNull_f(smallBean);
+    Map<String, SmallBean> map = new HashMap<>();
+    bean1.setChildMap(map);
+
+    Encoder<NestedSmallBeanWithNonNullField> encoder1 =
+            Encoders.bean(NestedSmallBeanWithNonNullField.class);
+    List<NestedSmallBeanWithNonNullField> beans1 = Arrays.asList(bean1);
+    Dataset<NestedSmallBeanWithNonNullField> ds1 = spark.createDataset(beans1, encoder1);
+
+    StructType schema = ds1.schema();
+    Assert.assertFalse(schema.fields()[schema.fieldIndex("nonNull_f")].nullable());
```

Comment:
how about `schema.apply("nonNull_f").nullable()`?

Reply:
yes, that is better. changed.